### PR TITLE
Run in a single test, create custom TMT results.yaml with useful logs/artifacts

### PR DESCRIPTION
**Pull request by martinpitt**
We are going to call rpminspect_runner.sh just once for a package update. All the extra result parsing and splitting is obsolete, it will be replaced with making rpminspect's new builtin viewer [1] easily accessible from TF results.
    
[1] https://github.com/rpminspect/rpminspect/tree/main/contrib

----
I configured https://quay.io/repository/rhn_engineering_mpitt/rpminspect?tab=tags to auto-build on pushes to my fork.

I tested this locally as much as possible with the corresponding [pipeline change](https://github.com/fedora-ci/rpminspect-pipeline/pull/64), replicating a recent [real-life run](https://artifacts.dev.testing-farm.io/4ca77f30-116d-41c6-b78b-4812ba2859ad/):
```
git clone -b custom-results-yaml https://github.com/martinpitt/rpminspect-pipeline
cd rpminspect-pipeline
cat <<EOF > env.yaml
PREVIOUS_TAG: f37-updates
TASK_ID: 94237776
RPMINSPECT_PROFILE_NAME: rawhide
REPOSITORY_URL: https://src.fedoraproject.org/rpms/cockpit-podman
CONFIG_BRANCH: rawhide
DEFAULT_RELEASE_STRING: fc37
GIT_COMMIT: f6b1abe13516b37569d2dcd9b718746e97e2e486
EOF

git fetch; git reset --hard origin/custom-results-yaml
rm -rf /var/tmp/tmt/*; tmt run --all --verbose -e @env.yaml
```

this ran, and as expected failed a few inspections. This has two initial custom logs:

```
$ cat /var/tmp/tmt/run-001/rpminspect/execute/data/rpminspect/data/results.yaml
- name: /rpminspect
  result: pass
  log:
    - data/rpminspect/output.txt
    - data/rpminspect/data/verbose.log
    - data/rpminspect/data/result.json
```

Now I want to test this end-to-end via TF, to see how that ends up in results.xml and how we can connect oculus and rpminspect's viewer.html. I hope/think that CI on this PR will do that.

I configured https://quay.io/repository/rhn_engineering_mpitt/rpminspect?tab=tags to auto-build on pushes to my fork.

 - [x] Wait for Testing Farm to get tmt 1.18.0 to support custom results.yaml (@thrix working on it)
 - [x] Teach gluetool-modules to consider all `log:` entries from results.yaml and expose them in results.xml: https://gitlab.com/testing-farm/gluetool-modules/-/merge_requests/282
 - [x] [rebase the above to testing-farm branch](https://gitlab.com/testing-farm/gluetool-modules/-/merge_requests/293)
 - [x] deploy current testing-farm branch (@thrix working on it)
 - [x] Rebase after some initial cleanups PR #83 and fixing rpminspect crash
 - [x] Reorganize oculus page to show plan and test logs/reproducer etc. on the bottom, with a jump link in the green bar: https://gitlab.com/testing-farm/oculus/-/merge_requests/29
 - [x] Add rpminspect viewer.html as log artifact
 - [x] Support HTML artifacts properly in TF viewer: https://gitlab.com/testing-farm/oculus/-/merge_requests/25

Follow-up/ignore:
 - [ ] Rework `rpminspect_zuul_runner.sh`, this would be completely broken right now.
 - [ ] Clean up `$RPMINSPECT_WORKDIR`. Not really used any more, but rpminspect_zuul_runner.sh scripts are full of it still

Corresponding pipeline changes are in https://github.com/fedora-ci/rpminspect-pipeline/pull/64 .

### CODE REVIEW

**Comment by martinpitt** [comment]
I sent a testing farm PR to plumb through the additional artifacts to results.xml and thus to the UI: https://gitlab.com/testing-farm/gluetool-modules/-/merge_requests/282 . This needs to land and get rolled out before we can make any progress here.

**Comment by martinpitt** [comment]
@msrb, @thrix : Hmm, it seems the environment files disappeared? My [original reference test run](https://artifacts.dev.testing-farm.io/32eccead-7105-4edb-bab3-4660fe895061/) as well as a [recent one](https://artifacts.dev.testing-farm.io/4ca77f30-116d-41c6-b78b-4812ba2859ad/) still exist, but the `tmt-environment-rpminspect.yaml` file mentioned in the reproducer does not exist any more.

It's plausible that the recent rpminspect-runner changes around workdir may have accidentally dropped this file from the artifacts, but why did it also *actively disappear* from an old result?

I'll try to guess how they should look like, so that I can locally run an rpminspect pipeline. This PR is finally unblocked after last Friday's TF release. :tada:

**Comment by martinpitt** [comment]
Ah, one can see the environment in the [tmt log](https://artifacts.dev.testing-farm.io/32eccead-7105-4edb-bab3-4660fe895061/work-rpminspect0a1Ftj/log.txt.gz), thus a working file looks like this:

```yaml
PREVIOUS_TAG: f37-updates
TASK_ID: 93912669
RPMINSPECT_PROFILE_NAME: rawhide
REPOSITORY_URL: https://src.fedoraproject.org/rpms/cockpit
CONFIG_BRANCH: rawhide
DEFAULT_RELEASE_STRING: fc37
GIT_COMMIT: 9fa3d64b44d624a37ab30ff25c56117cb31e86a9
```

**Comment by martinpitt** [comment]
Unfortunately that old cockpit test cannot be run any more, as `rpminspect` crashes: `rpminspect: libannocheck_init error: version number too small`. So I ran a pipeline for this [recent successful cockpit-podman bodhi result](https://artifacts.dev.testing-farm.io/4ca77f30-116d-41c6-b78b-4812ba2859ad/) through jenkins, and it ended up producing [this result](https://artifacts.dev.testing-farm.io/ca2bea76-d1cd-4818-b0a7-2365d52d6bfa/). This is almost good, it does have the new attachments, they are just in the wrong directory.

**Comment by msrb** [comment]
As discussed on IRC, we don't use the Zuul scripts for anything right now. So I would leave those alone -- no need to update them in this PR.

File: rpminspect_runner.sh [hunk 1]
```diff
@@ -164,98 +151,108 @@ get_before_module_build() {
...
+        --format=json \
+        --output="${output_filename}" \
+        --verbose \
+        ${arches:+--arches=$arches} \
+        ${default_release_string:+--release=$default_release_string} \
+        ${profile_name:+--profile=$profile_name} \
+        ${tests:+--tests=$tests} \
+        ${before_build} \
+        ${after_build_param} \
+        > $verbose_log 2>&1 || rc=$?
```
Comment (msrb) [inline rpminspect_runner.sh:219]: Nice trick with the `|| rc=$?` ;)

**Review by msrb — APPROVED** [review]

**Comment by msrb** [comment]
Removing so many lines of code -- amazing! :)

I'd say let's merge this and test the changes end-to-end in the pipeline.

**Pull request merged**